### PR TITLE
some more Float16 methods

### LINKED ISSUE #9939: rem not defined for Float16
This demonstrates the issue:
```
julia> rationalize(float16(1/3))
ERROR: rem not defined for Float16
 in divrem at number.jl:21
 in rationalize at rational.jl:117
 in rationalize at rational.jl:134
```

### PR DESCRIPTION
**Pull request by simonbyrne**
Fixes #9939 (and #9897).

### CODE REVIEW

File: base/float16.jl [hunk 1]
```diff
@@ -149,8 +149,16 @@ for func in (:sin,:cos,:tan,:asin,:acos,:atan,:sinh,:cosh,:tanh,:asinh,:acosh,
         $func(a::Complex32) = complex32($func(complex64(a)))
     end
 end
-atan2(y::Float16, x::Float16) = float16(atan2(float32(y), float32(x)))
-hypot(a::Float16, b::Float16) = float16(hypot(float32(a), float32(b)))
+
+for func in (:div,:fld,:cld,:rem,:mod,:mod1,:rem1,:fld1,
```
Comment (eschnett) [inline base/float16.jl]: The `*1` functions (`mod1`, `rem1`, `fld1`) are already defined generically for all `Real` types.

(The generic versions of `rem1` and `fld1` with `Real` arguments in `operators.jl` look suspicious -- I think they are wrong; their definition is probably only correct for integers.)

Do you want to add test cases for the new functions as well?

Comment (simonbyrne) [inline base/float16.jl]: Hmm, good point. I'll get rid of those for now.

Comment (simonbyrne) [inline base/float16.jl]: I added a test for the immediate issue. In general, `Float16` is pretty spares when it comes to tests, but perhaps that's a separate issue.

**Comment by eschnett** [comment]
This looks good.

**Comment by jiahao** [comment]
Ref: #5942

**Pull request merged**